src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -26,7 +26,13 @@ NonClientIslandWindow::NonClientIslandWindow(const ElementTheme& requestedTheme)
 {
 }
 
-NonClientIslandWindow::~NonClientIslandWindow() = default;
+void NonClientIslandWindow::Close()
+{
+    // Avoid further callbacks into XAML/WinUI-land after we've Close()d the DesktopWindowXamlSource
+    // inside `IslandWindow::Close()`. XAML thanks us for doing that by not crashing. Thank you XAML.
+    SetWindowLongPtr(_dragBarWindow.get(), GWLP_USERDATA, 0);
+    IslandWindow::Close();

```

Comment:
Without this, `NonClientIslandWindow` would crash here every single time I close a window:
https://github.com/microsoft/terminal/blob/c6215c8b510ff82a3c0891d5793e7a89e2a0ce10/src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp#L92